fatiando/gridder.py

```diff
@@ -439,3 +445,281 @@ def cut(x, y, scalars, area):
+        nps.append((int(numpy.ceil((npt[ii] - a.shape[ii])/2.)),
+                    int(numpy.floor((npt[ii] - a.shape[ii])/2.))))
+
+    # If it will be needed, compute the mean
+    meanneeded = ['lintaper', 'oddreflectiontaper']
+    if padtype.lower() in meanneeded:
+        m = numpy.mean(a)
+    # Use numpy's padding routines where possible
+    if _is_number(padtype):
+        # Pad with value
+        ap = numpy.pad(a, nps, mode='constant',
+                       constant_values=(float(padtype), float(padtype)))
+    elif padtype.lower() == 'mean':
+        # Pad with the mean
+        ap = numpy.pad(a, nps, mode='mean')
+    elif padtype.lower() == 'lintaper':
+        # Linearly taper to the mean
+        ap = numpy.pad(a, nps, mode='linear_ramp', end_values=(m, m))
+    elif padtype.lower() == 'edge':
+        # Pad with edge values
+        ap = numpy.pad(a, nps, mode='edge')
+    elif padtype.lower() == 'reflection':
+        # Pad with even reflection
+        ap = numpy.pad(a, nps, mode='reflect',
+                       reflect_type='even')
+    elif padtype.lower() == 'oddreflection':
+        # Pad with odd reflection
+        ap = numpy.pad(a, nps, mode='reflect',
+                       reflect_type='odd')
+    elif padtype.lower() == 'oddreflectiontaper':
+        # Pad with odd reflection and a cosine taper to mean
+        ap = (numpy.pad(a, nps, mode='reflect',
+                        reflect_type='odd') - m)
+        for ii in range(0, nd):
+            ap = numpy.apply_along_axis(_costaper, ii, ap, lp=nps[ii][0],
+                                        rp=nps[ii][1])
+        ap += m
+
+    return ap, nps
+
+
+def unpad_array(a, nps):
+    '''
+    Unpads an array using the outputs from pad_array.
+
+    This function takes a padded array and removes the padding from both.
+    Effectively, this is a complement to gridder.cut for when you already
+    know the number of elements to remove.
+
+    .. note:: Unlike :func:`~fatiando.gridder.pad_array`, this returns a slice
+        of the input array.  Therefore, any changes to the padded array will be
+        reflected in the unpadded array.
+
+    Parameters:
+
+    * a : N-D array
+        Array to be un-padded.  Can be of arbitrary dimension.
+    * nps : list
+        List of tuples giving the min and max indices for the cutoff.
+        Identical to nps returned by pad_array
+
+    Returns:
+
+    * b : N-D array
+        Array of same dimension as a, with padding removed
+
+    '''
+    o = []
+    for ii in range(0, a.ndim):
+        o.append(slice(nps[ii][0], a.shape[ii] - nps[ii][1]))
+    b = a[o]
+
+    return b
+
+
+def padcoords(xy, s, nps):
+    '''
+    Pads coordinate vectors.
+
+    Designed to be used in concert with :func:`~fatiando.gridder.pad_array`,
+    this function takes a list of coordinate vectors and pads them using the
+    same discretization.
+
+    Parameters:
+
+    * xy : List
+        List of arrays of coordinates
+    * s : tuple
```

Comment:
Also, could you rename this argument to `shape`? That's the standard numpy convention and is used everywhere, so it would be nice to maintain consistency.